src/chainstate/stacks/index/node.rs

```diff
@@ -569,18 +593,17 @@ impl TrieCursor {
 pub struct TrieLeaf {
     pub path: Vec<u8>,      // path to be lazily expanded
     pub data: MARFValue,    // the actual data
-    backptr: TriePtr        // pointer back to the previous version of this leaf
 }
 
 impl fmt::Debug for TrieLeaf {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "TrieLeaf(path={} data={} backptr={:?})", &to_hex(&self.path), &to_hex(&self.data.to_vec()), &self.backptr)
+        write!(f, "TrieLeaf(path={} data={})", &to_hex(&self.path), &to_hex(&self.data.to_vec()))
     }
 }
 
 impl PartialEq for TrieLeaf {
     fn eq(&self, other: &TrieLeaf) -> bool {
-        self.path == other.path && slice_partialeq(self.data.as_bytes(), other.data.as_bytes()) && self.backptr == other.backptr
+        self.path == other.path && slice_partialeq(self.data.as_bytes(), other.data.as_bytes())
```

Comment:
I don't think this is right.  Two leaves are different if they're inserted in different blocks, even if they have the same path and value.  i.e. they'll be expected to have different MARF merkle proofs.